### ISSUE TITLE
[Feature Request] Fuzzy name search to taxonomy identifier 

### ISSUE THREAD

**Issue by jolespin** [apcamargo/taxopy#15]
Similar to https://github.com/etetoolkit/ete/blob/1582ea2aa0d28065f4757b8b5af74367f6abe19f/ete4/ncbi_taxonomy/ncbiquery.py#L112C30-L112C30

```
    def get_fuzzy_name_translation(self, name, sim=0.9):
        """Return taxid, species name and match score from the NCBI database.

        The results are for the best match for name in the NCBI
        database of taxa names, with a word similarity >= `sim`.

        :param name: Species name (does not need to be exact).
        :param 0.9 sim: Min word similarity to report a match (from 0 to 1).
        """
```

For example, EukZoo has an annotation from a source organism id `AddRef0031` labeled as species `Paramecium tetraurelia` and strain `Stock d4-2`.  A manual search for this shows  https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=412030 but the keyword "stock" is missing even though they are certainly the same.  

Would it be possible to include this type of searching?

**Comment by apcamargo** [comment]
Hi @jolespin,

That's a good idea. I could implement a Levenshtein distance search for that. It might not be super fast because I don't want to add more dependencies, so the computation would have to be pure Python.

I don't think I'll have the time to implement that soon. If you need this functionality, you could try to implement that on your end. The https://github.com/maxbachmann/python-Levenshtein package seems like a good choice to identify similar names. You'd just need to loop though all names and compare each one with your query.

**Comment by jolespin** [comment]
I agree, if this package makes installation a pain then it's not worth it. If you think the fuzzy search would be useful it could be a good idea to do what pandas did with the excel support for reading/writing xlsx.  It's not included in the base installation and if a user tries to read excel docs then it provides an error telling the user to install the package if they want to use the functionality.

**Comment by apcamargo** [comment]
Can you point me to the exact error that pandas provides? I want to understand what is the proper way of implementing these error messages.

I can use [optional dependencies](https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#optional-dependencies) to have an alternative version of taxopy with fuzzy name search support, like this:
```
pip install taxopy[fuzzy-search]
```

**Comment by jolespin** [comment]
Here you go: 

```
Python 3.10.14 | packaged by conda-forge | (main, Mar 20 2024, 12:53:34) [Clang 16.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pandas as pd
>>> pd.read_excel("/Users/jolespin/Cloud/Informatics/Development/Packages/veba/SOURCES.xlsx")
Traceback (most recent call last):
  File "/Users/jolespin/miniconda3/envs/test_env/lib/python3.10/site-packages/pandas/compat/_optional.py", line 135, in import_optional_dependency
    module = importlib.import_module(name)
  File "/Users/jolespin/miniconda3/envs/test_env/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'openpyxl'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jolespin/miniconda3/envs/test_env/lib/python3.10/site-packages/pandas/io/excel/_base.py", line 495, in read_excel
    io = ExcelFile(
  File "/Users/jolespin/miniconda3/envs/test_env/lib/python3.10/site-packages/pandas/io/excel/_base.py", line 1567, in __init__
    self._reader = self._engines[engine](
  File "/Users/jolespin/miniconda3/envs/test_env/lib/python3.10/site-packages/pandas/io/excel/_openpyxl.py", line 552, in __init__
    import_optional_dependency("openpyxl")
  File "/Users/jolespin/miniconda3/envs/test_env/lib/python3.10/site-packages/pandas/compat/_optional.py", line 138, in import_optional_dependency
    raise ImportError(msg)
ImportError: Missing optional dependency 'openpyxl'.  Use pip or conda to install openpyxl.
```

**Comment by apcamargo** [comment]
I started working on this feature in this branch: https://github.com/apcamargo/taxopy/tree/fuzzy-matching

**Comment by apcamargo** [comment]
This functionality is now included in release 0.13.0.
